gui/gui-manager.cpp

```diff
@@ -547,10 +547,16 @@ void GuiManager::runLoop() {
 
 		redraw();
 
-		// Delay until the allocated frame time is elapsed to match the target frame rate
-		uint32 actualFrameDuration = _system->getMillis(true) - frameStartTime;
-		if (actualFrameDuration < targetFrameDuration) {
-			_system->delayMillis(targetFrameDuration - actualFrameDuration);
+		// Delay until the allocated frame time is elapsed to match the target frame rate.
+		// In case we have vsync enabled, we force a minimum frame duration of 1 millisecond
+		// since otherwise, we'd have a very sluggish cursor on 60Hz displays.
+		if (g_system->getFeatureState(OSystem::kFeatureVSync)) {
+			_system->delayMillis(1);
```

Comment:
why add 1ms? vsync will limit

Reply:
In case I don't set the additional 1ms, CPU utilization of one core will stay at 100% all the time with vsync enabled. Not entirely sure why this happens though :/

Reply:
1ms is needed in case no screen changes and OpenGL backend will not call GL swap buffers (with vsync)

Reply:
can you add comments about why limit to 1ms?